File: 02_selection_sort/swift/01_selection_sort.swift
func selectionSort(arr: [Int]) -> [Int] {
	let n = arr.count
	var newarr = arr
	
	for i in 0..<n-1 {
		var min_idx = i
		for j in i+1..<n { 
			if newarr[j] < newarr[min_idx] {
				min_idx = j
			}
		}
		let temp = newarr[min_idx] 
		newarr[min_idx] = newarr[i]
		newarr[i] = temp
	}
	return newarr
}

var myarr = [5, 3, 6, 2, 10]
print(selectionSort(arr: myarr))

File: 02_selection_sort/javascript/01_selection_sort.js
function selectionSort(arr){
	n = arr.length;

	for(let i = 0; i < n-1; ++i){
		let min_idx = i;
		for(let j = i+1; j < n; ++j){
			if (arr[j] < arr[min_idx]) 
				min_idx = j; 
		}
		let temp = arr[min_idx]; 
		arr[min_idx] = arr[i]; 
		arr[i] = temp;
	}
	
	return arr;
}

var arr = [5, 3, 6, 2, 10];
console.log(selectionSort(arr));